src/NHibernate/Util/ParameterHelper.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using NHibernate.Engine;
+using NHibernate.Proxy;
+using NHibernate.Type;
+
+namespace NHibernate.Util
+{
+	internal static class ParameterHelper
```

Comment:
The methods were extracted from `AbstractQueryImpl` in order to be used by `ConstantTypeLocator`.